### PR TITLE
When building Julia, pass `TAGGED_RELEASE_BANNER=...` to `make`; also, use a `Make.user` file to pass the options to `make`

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@staticfloat How do I go about interpolating these `make` flags correctly, in a way that preserves the `"` at the beginning and end of the banner string?

**Comment by DilumAluthge** [comment]
Should we just `cat` our desired flags to `Make.user`?

**Comment by DilumAluthge** [comment]
@ararslan Once Elliot has approved this, I'll merge it, and that will take care of the tagged release banners.

**Comment by DilumAluthge** [comment]
@staticfloat Do we need `VERBOSE=1`? I saw it in the Buildbot configuration, so I added it here, but I'm not sure if it's actually useful.

**Comment by DilumAluthge** [comment]
Also, do I need to put `JULIA_CPU_TARGET=...` in the `Make.user` file

File: pipelines/main/platforms/build_linux.yml [hunk 1]
```diff
@@ -27,10 +27,33 @@ steps:
...
+          rm -f Make.user.tmp
+          echo "${MAKE_FLAGS?}" | tr "," "\n" > Make.user.tmp
+          if [[ "$${BUILDKITE_PIPELINE_SLUG:?}" == "julia-master" ]]
+          then
+            banner="Official https://julialang.org/ release"
+            echo "TAGGED_RELEASE_BANNER=\"$${banner:?}\"" >> Make.user.tmp
+          elif [[ "$${BUILDKITE_PIPELINE_SLUG:?}" =~ ^julia-release- ]]
+          then
+            banner="Official https://julialang.org/ release"
+            echo "TAGGED_RELEASE_BANNER=\"$${banner:?}\"" >> Make.user.tmp
```
Comment (staticfloat) [inline pipelines/main/platforms/build_linux.yml:40]:
```suggestion
          if [[ "$${BUILDKITE_PIPELINE_SLUG:?}" == "julia-master" ]] || [[ "$${BUILDKITE_PIPELINE_SLUG:?}" =~ ^julia-release- ]]; then
            banner="Official https://julialang.org/ release"
            echo "TAGGED_RELEASE_BANNER=\"$${banner:?}\"" >> Make.user.tmp
```

File: pipelines/main/platforms/build_linux.yml [hunk 2]
```diff
@@ -27,10 +27,33 @@ steps:
...
+          else
+            :
+          fi
+          echo "VERBOSE=1" >> Make.user.tmp
+          rm -f Make.user
+          awk NF Make.user.tmp > Make.user
+          rm -f Make.user.tmp
+          wc -l Make.user
+          cat Make.user
```
Comment (staticfloat) [inline pipelines/main/platforms/build_linux.yml:53]: I'm confused, why are you doing this?  Why not just write out to `Make.user` directly?

Comment (DilumAluthge) [reply pipelines/main/platforms/build_linux.yml:53]: I wanted to remove blank lines, and I couldn't figure out how to do that in place.

Comment (staticfloat) [reply pipelines/main/platforms/build_linux.yml:48]: I..... I don't think it's that important.  😂

File: pipelines/main/platforms/build_linux.yml [hunk 3]
```diff
@@ -27,10 +27,25 @@ steps:
           # export JULIA_CPU_TARGET, which is picked up by the build system
           ARCH="$${ARCH??}" source .buildkite/utilities/export_julia_cpu_target.sh
 
+          echo "--- Create the Make.user file"
+          rm -f Make.user
```
Comment (staticfloat) [inline pipelines/main/platforms/build_linux.yml:31]:
```suggestion
```

File: pipelines/main/platforms/build_linux.yml [hunk 4]
```diff
@@ -27,10 +27,25 @@ steps:
...
+          echo "${MAKE_FLAGS?}" | tr "," "\n" > Make.user
+          if [[ "$${BUILDKITE_PIPELINE_SLUG:?}" == "julia-master" ]] || [[ "$${BUILDKITE_PIPELINE_SLUG:?}" =~ ^julia-release- ]]; then
+            banner="Official https://julialang.org/ release"
+            echo "TAGGED_RELEASE_BANNER=\"$${banner:?}\"" >> Make.user
+          elif [[ "$${BUILDKITE_PIPELINE_SLUG:?}" == "julia-buildkite" ]]
+          then
```
Comment (staticfloat) [inline pipelines/main/platforms/build_linux.yml:37]:
```suggestion
          elif [[ "$${BUILDKITE_PIPELINE_SLUG:?}" == "julia-buildkite" ]]; then
```

File: pipelines/main/platforms/build_linux.yml [hunk 5]
```diff
@@ -27,10 +27,25 @@ steps:
...
+            banner="Not an official binary; built as part of the CI infrastructure testing"
+            echo "TAGGED_RELEASE_BANNER=\"$${banner:?}\"" >> Make.user
+          else
+            :
```
Comment (staticfloat) [inline pipelines/main/platforms/build_linux.yml:41]:
```suggestion
```

File: pipelines/main/platforms/build_linux.yml [hunk 6]
```diff
@@ -27,10 +27,25 @@ steps:
...
+          fi
+          echo "VERBOSE=1" >> Make.user
+          wc -l Make.user
```
Comment (staticfloat) [inline pipelines/main/platforms/build_linux.yml:44]:
```suggestion
```

**Review by staticfloat — APPROVED** [review]

**Pull request merged**